### PR TITLE
feat: Hooks on pool page

### PR DESCRIPTION
**Pull request by garethfuller**
- [x] Hook icon in header
- [x] Hook contract address in contracts section

Screenshot 2024-10-31 at 16 18 26
Screenshot 2024-10-31 at 16 18 30

### CODE REVIEW

File: packages/lib/modules/pool/PoolDetail/PoolHeader/PoolMetaBadges.tsx [hunk 1]
```diff
@@ -46,6 +50,13 @@ export default function PoolMetaBadges() {
       <BalBadge color="font.secondary" fontSize="xs" textTransform="none">
         {getPoolTypeLabel(pool.type)}
       </BalBadge>
+      {hasHook && (
+        <BalBadge color="font.primary" fontSize="xs" h={8} w={8}>
```
Comment (agualis) [inline packages/lib/modules/pool/PoolDetail/PoolHeader/PoolMetaBadges.tsx:54]: I would add a popover like we have in the swap fee icon: 

hookTooltip

If it's easy we could show a link to the name of the hook or a link to the hook contract in the popover. (can be done in the future)

File: packages/lib/modules/hooks/useHook.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Address } from 'viem'
+import { Pool } from '../pool/PoolProvider'
+
+export function useHook(pool: Pool) {
+  const hasHook = !!pool.hook?.address
```
Comment (agualis) [inline packages/lib/modules/hooks/useHook.ts:5]: Out of curiosity: cannot we have more than one hook in a pool?

File: packages/lib/modules/pool/PoolDetail/PoolHeader/PoolMetaBadges.tsx (hunk 1 above)
Comment (garethfuller) [reply packages/lib/modules/pool/PoolDetail/PoolHeader/PoolMetaBadges.tsx:54]: Yeah metadata is coming in another PR.

File: packages/lib/modules/hooks/useHook.ts (hunk 2 above)
Comment (garethfuller) [reply packages/lib/modules/hooks/useHook.ts:5]: Nope.

**Pull request merged**